.github/workflows/gh-pages.yml

```diff
@@ -25,6 +25,15 @@ jobs:
         run: |
           git config --local user.name ${{ github.actor }}
           git config --local user.email "${{ github.actor }}@users.noreply.github.com"
+          # Preserve benchmark results, if any
```

Comment:
How is this related to benchmarks? Is it because its' the only ones that could be generated by cabal but not checked in?

Reply:
The [benchmarks CI](https://github.com/IntersectMBO/cardano-ledger/actions/workflows/bench.yml) is triggered at the same time as the Haddock one, but finishes first. It saves its [results](https://cardano-ledger.cardano.intersectmbo.org/dev/bench/) into a directory called `dev/bench` and commits that to the `gh-pages` branch without overwriting it. However, when the Haddocks CI finishes it force-pushes the `gh-pages` branch and loses the benchmark results. However, we don't want to keep the commits from the previous Haddocks run because then we'd be filling up our git repo with useless history. This tricky little bit of code gets the `dev` directory from the previous state of the `gh-pages` branch without bringing in its other history.